docs/book/utils/markerdocs/html.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"html"
+	"io"
+
+)
+
+// NB(directxman12): we use this instead of templates to avoid
+// weird issues with whitespace in elements rendered as inline.
+// Writing with templates was getting tricky to do without
+// compromising readability.
```

Comment:
Our use-case (rendering html using json) is sort of the sweet spot for using templates, but Go templates doesn't make it easy.

Reply:
sort-of, but not quite in our case :-(